src/components/app_catalog/app_list/list_inner.js

```diff
@@ -0,0 +1,178 @@
+import { replace } from 'connected-react-router';
+import AppListItems from './list_items';
+import AppListSearch from './list_search';
+import lunr from 'lunr';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+class AppListInner extends React.Component {
+  state = {
+    filters: [],
+    searchQuery: null,
+    iconErrors: {},
+  };
+
+  // Contains refs to all the app-container divs in dom so that we can
+  // scroll to them if needed.
+  appRefs = {};
+
+  index = null;
+
+  constructor(props) {
+    super(props);
+
+    this.index = lunr(function() {
+      this.ref('name');
+      this.field('name');
+      this.field('description');
+      this.field('keywords');
+
+      const apps = Object.values(props.catalog.apps).map(appVersions => {
+        return appVersions.sort((a, b) => {
+          return new Date(b.created) - new Date(a.created);
+        })[0];
+      });
+
+      for (const app of apps) this.add(app);
+    });
+  }
```

Comment:
I actually forgot about this, sorry.
What I was trying to propose is dropping `lunr` from this altogether, since it's not providing an actual performance benefit. What do you think about that?

Reply:
Lunr is providing a UX benefit. What are you going to replace it with?

Reply:
I've replaced lunr with a simpler algorithm. Now search times are comparable to before, but with a much lower CPU usage